pages/HomeScreen.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import styled from "styled-components/native";
+import { Image, ActivityIndicator } from "react-native";
+
+import { Input } from "../components/Input";
+import { Header } from "../components/Header";
+
+export const HomeScreen = ({ navigation: { navigate } }) => {
+  const [books, setBooks] = useState([]);
+  const [searchText, setSearchText] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  const addText = () => {
+    setLoading(true);
+    fetch(
+      `https://www.googleapis.com/books/v1/volumes?maxResults=40&q=${searchText}`
+    )
+      .then(res => res.json())
+      .then(json => {
+        setBooks(json.items);
+        setSearchText("");
+        setLoading(false);
+      })
+      .catch(() => {
+        console.log(error.code);
+      });
+  };
+
+  return (
+    <ContainerNoScroll>
+      <Header title="BOOKIFY" />
+      <Input
+        addText={addText}
+        searchText={searchText}
+        setSearchText={setSearchText}
+      />
+      {loading && (
+        <ActivityIndicator size="large" color="#00bfff" animating={true} />
+      )}
+      {!loading && (

```

Comment:
you could also render this with the ternary logic.
Both works fine.
`{loading ? <ActivityIndicator> : <BooksContainer />}`